dockertest/output/dockerinfo.py

```diff
@@ -0,0 +1,104 @@
+"""
+Handler for 'docker info'. Runs command, returns output in
+machine-friendly form.
+"""
+
+import collections
+import subprocess
+
+
+class DockerInfo(collections.MutableMapping):
+    """
+    Parser of 'docker info' output
+    """
+
+    def __init__(self, info_string=None, docker_path=None):
+        if info_string is None:
+            if docker_path is None:
+                docker_path = 'docker'
+            info_string = subprocess.check_output(docker_path + ' info',
+                                                  shell=True,
+                                                  close_fds=True)
```

Comment:
Blah, terrible choice of words...let me explain another way...

Having ``info_string`` optional and ``_build_table()`` private seems to defeat the purpose of defining a new type of thing.  If a class is used, then obtaining the data, and processing it, and presenting it shouldn't be tightly-coupled to creation.  It makes it very difficult to sub-class w/o re-writing most of the entire thing, so why have a class in the first place?

e.g. If I want to subclass ``DockerInfo`` to parse the output in a different way, because of different command-line options, I'd have to override ``__init__()``, `` _build_table()``, and maybe even ``sub_fields()``.  Basically the entire class.

(Small Mountain):  I'm not in a hurry, but if you are...maybe this "class" should really just be a function?

(Challenging Expedition):  To grow, and become powerful in the ways of the For^H^H^H OOP...the presentation-aspect shouldn't be tightly-coupled to both creation and processing.  Kind of like a **miniature** version of the MVC-like patterns: [MVP](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter) or [MVA](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93adapter)

e.g.: (one possibility)

```python

class DockerInfo(object):
    """
    Parser of 'docker info' output
    """

    def __init__(self, info_string):
        self.info_string = info_string  # The model

    @classmethod
    def MakeDockerInfo(cls, docker_path=None):
        return cls(subprocess.check_output...))

    def build_table(self):  # The adapter or presenter
        ...

    def __getitem__(self, key):  # The view
        ...
```

Now the data is close to all operations, yet I'm free to subclass and alter any individual aspect w/o needing to completely re-write the others.

Does that help/make more sense?